src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -75,14 +76,20 @@ public void createBuildPlanForExercise(ProgrammingExercise exercise, String plan
         jenkinsBuildPlanService.createBuildPlanForExercise(exercise, planKey, repositoryURL, testRepositoryURL);
     }
 
-    /**
-     * Auxiliary repositories are not supported for Gitlab/Jenkins configurations.
-     *
-     * @param exercise for which the build plans should be recreated
-     */
     @Override
     public void recreateBuildPlansForExercise(ProgrammingExercise exercise) {
-        // Auxiliary repositories are currently not supported for Gitlab/Jenkins configurations.
+        final String projectKey = exercise.getProjectKey();
+
+        if (!jenkinsBuildPlanService.projectFolderExists(projectKey)) {
+            createProjectForExercise(exercise);
+        }
+
+        deleteBuildPlan(exercise.getProjectKey(), exercise.getTemplateBuildPlanId());
+        deleteBuildPlan(exercise.getProjectKey(), exercise.getSolutionBuildPlanId());
```

Comment:
```suggestion
        deleteBuildPlan(projectKey, exercise.getTemplateBuildPlanId());
        deleteBuildPlan(projectKey, exercise.getSolutionBuildPlanId());
```
[nit] would make the code a bit shorter